eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -67,10 +89,25 @@ public ClusteringMongoReader(MongoTemplate mongoTemplate, String assembly, int c
 
     @Override
     public SubmittedVariantEntity read() {
+        return retryTemplate.execute(retryContext -> doRead());
+    }
+
+    public SubmittedVariantEntity doRead() {
         // Read from dbsnp collection first and subsequently EVA collection
-        Document nextElement = (dbsnpCursor != null && dbsnpCursor.hasNext()) ? dbsnpCursor.next() :
-                (evaCursor.hasNext() ? evaCursor.next() : null);
-        return (nextElement != null) ? getSubmittedVariantEntity(nextElement) : null;
+        Document nextElement = null;
+        if (dbsnpCursor != null && dbsnpCursor.hasNext()) {
+            currentCollection = DBSNP_COLLECTION;
+            nextElement = dbsnpCursor.next();
+        } else if (evaCursor.hasNext()) {
```

Comment:
evaCursor can be null too. So suggest adding "evaCursor != null" here..

Reply:
`dbsnpCursor` can be `null` because we set it to null. I don't think `evaCursor` can be null

Reply:
I think Tim is right that `evaCursor` should never be null, but I guess adding the check doesn't hurt...